src/core/document.js

```diff
@@ -351,319 +372,302 @@ var PDFDocument = (function PDFDocumentClosure() {
-        // as well as any existing well-formed custom entries.
-        for (let key of infoDict.getKeys()) {
-          const value = infoDict.get(key);
-
-          if (DocumentInfoValidators[key]) {
-            // Make sure the (standard) value conforms to the specification.
-            if (DocumentInfoValidators[key](value)) {
-              docInfo[key] = (typeof value !== 'string' ?
-                              value : stringToPDFString(value));
-            } else {
-              info(`Bad value in document info for "${key}".`);
-            }
-          } else if (typeof key === 'string') {
-            // For custom values, only accept white-listed types to prevent
-            // errors that would occur when trying to send non-serializable
-            // objects to the main-thread (for example `Dict` or `Stream`).
-            let customValue;
-            if (isString(value)) {
-              customValue = stringToPDFString(value);
-            } else if (isName(value) || isNum(value) || isBool(value)) {
-              customValue = value;
-            } else {
-              info(`Unsupported value in document info for (custom) "${key}".`);
-              continue;
-            }
-
-            if (!docInfo['Custom']) {
-              docInfo['Custom'] = Object.create(null);
-            }
-            docInfo['Custom'][key] = customValue;
-          }
+
+      if (found) {
+        stream.skip(9);
+        let ch;
+        do {
+          ch = stream.getByte();
+        } while (isSpace(ch));
+        let str = '';
+        while (ch >= 0x20 && ch <= 0x39) { // < '9'
+          str += String.fromCharCode(ch);
+          ch = stream.getByte();
         }
-      }
-      return shadow(this, 'documentInfo', docInfo);
-    },
-    get fingerprint() {
-      var xref = this.xref, hash, fileID = '';
-      var idArray = xref.trailer.get('ID');
-
-      if (Array.isArray(idArray) && idArray[0] && isString(idArray[0]) &&
-          idArray[0] !== EMPTY_FINGERPRINT) {
-        hash = stringToBytes(idArray[0]);
-      } else {
-        if (this.stream.ensureRange) {
-          this.stream.ensureRange(0,
-            Math.min(FINGERPRINT_FIRST_BYTES, this.stream.end));
+        startXRef = parseInt(str, 10);
+        if (isNaN(startXRef)) {
+          startXRef = 0;
         }
-        hash = calculateMD5(this.stream.bytes.subarray(0,
-          FINGERPRINT_FIRST_BYTES), 0, FINGERPRINT_FIRST_BYTES);
       }
+    }
+    return shadow(this, 'startXRef', startXRef);
+  }
 
-      for (var i = 0, n = hash.length; i < n; i++) {
-        var hex = hash[i].toString(16);
-        fileID += hex.length === 1 ? '0' + hex : hex;
+  // Find the header, get the PDF format version and setup the
+  // stream to start from the header.
+  checkHeader() {
+    const stream = this.stream;
+    stream.reset();
+
+    if (!find(stream, '%PDF-', 1024)) {
+      // May not be a PDF file.
```

Comment:
Given the old comment `May not be a PDF file, continue anyway.`, how about adjusting this one to something like `May not be a PDF file, but don't throw an error and let parsing continue.` instead?